ragengine/vector_store/faiss_store.py

```diff
@@ -0,0 +1,228 @@
+            self._persist(index_name)
+
+    def update_document(self, document: Document, index_name: str):
+        # Updates an existing document in the FAISS vector store.
+        if index_name not in self.index_map:
+            raise ValueError(f"No such index: '{index_name}' exists.")
+        llama_doc = LlamaDocument(text=document.text, metadata=document.metadata, id_=document.doc_id)
+        try:
+            self.index_map[index_name].update_ref_doc(llama_doc)
+        except NotImplementedError as e:
+            print("Update is equivalent to deleting the document and then inserting it again.")
+            print(f"Update not yet fully implemented for index. Skipping document {document.doc_id}. Error: {e}")
+        except Exception as e:
+            print(f"Unable to Update Document in the VectorStoreIndex. Skipping. Error: {e}")
+        finally:
+            self._persist(index_name)
+
+    def refresh_documents(self, documents: List[Document], index_name: str) -> List[bool]:
+        # Updates existing documents and inserts new documents in the vector store.
+        if index_name not in self.index_map:
+            raise ValueError(f"No such index: '{index_name}' exists.")
+
+        llama_docs = [LlamaDocument(text=doc.text, metadata=doc.metadata, id_=doc.doc_id) for doc in documents]
+        try:
+            refresh_results = self.index_map[index_name].refresh_ref_docs(llama_docs)
+            # Returns a list of booleans indicating whether each document was successfully refreshed.
+            return refresh_results
+        except NotImplementedError as e:
+            print(f"Refresh is equivalent to insertion and update, which is equivalent to deletion and insertion.")
+            print(f"Refresh not yet fully implemented for index '{index_name}'. Error: {e}")
+        except Exception as e:
+            print(f"Unable to Refresh Documents in the VectorStoreIndex for index '{index_name}'. Error: {e}")
+        finally:
+            self._persist(index_name)
+    """
+
+    def list_all_indexed_documents(self) -> Dict[str, VectorStoreIndex]:
+        """Lists all documents in the vector store."""
+        return self.index_map
+
+    def document_exists(self, index_name: str, doc_id: str) -> bool:
+        """Checks if a document exists in the vector store."""
+        if index_name not in self.index_map:
+            print(f"No such index: '{index_name}' exists in vector store.")
+            return False
+        return doc_id in self.index_map[index_name].ref_doc_info
+
+    def _load_index_store(self):
+        """Loads the global SimpleIndexStore from disk."""
+        store_path = os.path.join(PERSIST_DIR, "store.json")
+
+        if not os.path.exists(store_path):
+            raise ValueError("No persisted index store found.")
+
+        # Load the global index store from the persisted JSON
+        self.index_store = SimpleIndexStore.from_persist_path(store_path)
+
+    def _load_indices(self):
```

Comment:
If we don't plan to support loading index from local disk in this release, remove the related code.

Reply:
Currently persist is the only function called every "index" call. It just saves db to a file, load is never called

Reply:
removed load code